aikido_perception/src/AprilTagsModule.cpp

```
@@ -0,0 +1,170 @@
+void AprilTagsModule::detectObjects(std::vector<dart::dynamics::SkeletonPtr>& skeleton_list,double _timeout, ros::Time timestamp)
+{
+	bool transform_lookedup = false;
+	//Looks at all detected tags, looks up config file 
+	//Appends new skeletons to skeleton list
+	do{
+		visualization_msgs::MarkerArrayConstPtr marker_message
+	 			= ros::topic::waitForMessage<visualization_msgs::MarkerArray>(mMarkerTopic,mNode,ros::Duration(_timeout));
+
+		for (const auto& marker_transform : marker_message->markers)
+		{
+			//Get marker, tag ID, and detection transform name 
+			const auto& marker_stamp = marker_transform.header.stamp;
+			const auto& tag_name = marker_transform.ns;
+			const auto& detection_frame = marker_transform.header.frame_id;
+
+			if(marker_stamp < timestamp){
+				dtwarn << "Marker stamp " << marker_stamp<< " for " << tag_name 
+				<< "is behind timestamp parameter: " <<timestamp <<"\n";
+
+				continue;
+			}
+
+			std::string skel_name;
+			Eigen::Isometry3d skel_offset;
+			dart::common::Uri skel_resource;
+
+			//check if tag name is in database
+			if (mConfigData->getTagNameOffset(tag_name,skel_name,skel_resource,skel_offset))
+			{
+
+				//Get orientation of marker
+				Eigen::Isometry3d marker_pose = aikido::perception::convertROSPoseToEigen(marker_transform.pose);
+
+
+				//For the frame-frame transform
+				tf::StampedTransform transform;
+
+				try{
+					mListener->waitForTransform(mReferenceFrameId,detection_frame,
+						marker_stamp,ros::Duration(_timeout));
+
+					mListener->lookupTransform(mReferenceFrameId,detection_frame,
+						marker_stamp, transform);
+				}
+				catch(const tf::ExtrapolationException& ex){
+					ROS_ERROR("%s",ex.what());
+					continue;
+				}
+				catch(const tf::TransformException& ex){
+					ROS_ERROR("%s",ex.what());
+					throw std::runtime_error("TF Error which cannot be recovered");
```

I don't think there is any point in catching this exception. I would just let it bubble up and require the user to handle it.